lib/components/add_expense/add_expense_form.dart

```diff
@@ -0,0 +1,157 @@
+                        DateFormat.yMd().format(date);
+                  }),
+            ]),
+            // Cost field
+            ExpenseFormField(
+              enabled: true,
+              maxCharacters: null,
+              keyboardType:
+                  const TextInputType.numberWithOptions(decimal: true),
+              inputFormatter: CurrencyTextInputFormatter.simpleCurrency(
+                  enableNegative: false),
+              controller: widget.controllerMap[amountTextFormFieldLabel]!,
+              labelText: amountTextFormFieldLabel,
+              hintText: amountTextFormFieldHint,
+              icon: null,
+              onSaved: (value) {},
+              validator: (value) {
+                return checkEmptyInput(value);
+              },
+            ),
+            // Description field
+            ExpenseFormField(
+              enabled: true,
+              maxCharacters: null,
+              keyboardType: TextInputType.text,
+              inputFormatter: FilteringTextInputFormatter.singleLineFormatter,
+              controller: widget.controllerMap[descriptionTextFormFieldLabel]!,
+              labelText: descriptionTextFormFieldLabel,
+              hintText: descriptionTextFormFieldHint,
+              icon: null,
+              onSaved: (value) {},
+              validator: (value) {
+                return checkEmptyInput(value);
+              },
+            ),
+            // Category field
+            ExpenseFormDropdown(
+              options: expenseCategories,
+              labelText: categoryTextFormFieldLabel,
+              controller: widget.controllerMap[categoryTextFormFieldLabel]!,
+              validator: (value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please select a value';
+                }
+                return null;
+              },
+              hintText: categoryTextFormFieldHint,
+              icon: null,
+            ),
+            ElevatedButton(
+              onPressed: () {
+                // Validate returns true if the form is valid, or false otherwise.
+                if (_formKey.currentState!.validate()) {
+                  ExpenseData expense = ExpenseData(
+                    id: "01234",
+                    description: widget
+                        .controllerMap[descriptionTextFormFieldLabel]!.text,
+                    cost: double.parse(widget
+                        .controllerMap[amountTextFormFieldLabel]!.text
+                        .replaceFirst("\$", "")
+                        .replaceAll(",", "")),
+                    date: widget.controllerMap[dateTextFormFieldLabel]!.text,
+                    category:
+                        widget.controllerMap[categoryTextFormFieldLabel]!.text,
+                    person:
+                        widget.controllerMap[personTextFormFieldLabel]!.text,
+                  );
+
+                  print(expense.toString());
```

Comment:
Can't remember if I added logger.dart at some point, but use that instead of print